rootfs/etc/nginx/template/nginx.tmpl

```diff
@@ -1092,6 +1092,9 @@ stream {
             set $global_rate_limit_exceeding n;
 
             {{ buildOpentracingForLocation $all.Cfg.EnableOpentracing $location }}
+            {{ if not $all.Cfg.OpentracingTrustIncomingSpan }}
```

Comment:
Is there a reason for this configuration to be global instead of location specific / per annotation?

Reply:
Nope. It could be added to [buildOpentracingForLocation()](https://github.com/kubernetes/ingress-nginx/blob/b1c8e3047ba31d7ea78f6e0915d187db75230ba5/internal/ingress/controller/template/template.go#L1346) and use the same logic as `isOTSetInLoc && isOTEnabledInLoc` if that's preferred, so there can be both global and location-specific settings.